### ISSUE TITLE
1.16.4: Chat color permissions, negating perm from OP for bypassing rules (luckperms)

### ISSUE THREAD

**Issue by Managering** [kangarko/ChatControl-Pro#2202]
Hi,

Question one:
I've been trying to disallow chatcolor using &1 etc with all the permission nodes I could find on chatcontrol. I ended up using a filter to strip the &* from messages. Not ideal.
Could you advice me which perms to negate to disallow this?
On the opposite side, I am trying to give a ranked group the permission to use /chatcontrol color. I've read quite some already asked questions that I could find on this, but simply put, none of the permissions work for me. What should it be?
We do use Luckperms.

Additionally I've been trying to negate the bypass on chat.txt for myself as OP. I tried luckperms false, I tried the -perm node for it. Neither work. Any advice?

**Comment by kangarko** [comment]
Hi!

For your first question, you can just give negative chatcontrol.color.dark_blue or any other https://helpch.at/docs/1.13/index.html?org/bukkit/class-use/ChatColor.html variant.

To let players use /chc color, give them chatcontrol.command.color permissions and also chatcontrol.gui.color.{name} permissions for the colors from the link above you want them to use.

Lastly, try giving yourself negative chatcontrol.bypass.rules.chat permission.

**Comment by Managering** [comment]
Hi,

https://imgur.com/a/2ykRMwE
First one doesn't work. Added all color codes as a false in Luckperms.

Tried the permissions for chc color too. Doesn't work. No permissions for command.

Tried to give myself a negative bypass permission. Tried to use luckperms with a false setting to it. Doesn't work either.

Basically none of these permissions work for me.

Additionally, I just found out hexcodes are possible too. Would like a way to block those :)
